packages/react-components/react-motion/library/src/factories/createPresenceComponent.ts

```diff
@@ -244,5 +247,20 @@ export function createPresenceComponent<MotionParams extends Record<string, Moti
       // Always normalize it to a function to simplify types
       [MOTION_DEFINITION]: typeof value === 'function' ? value : () => value,
     },
+    {
+      // Wrap `enter` in its own motion component as a static method, e.g. <Fade.In>
+      In: createMotionComponent(
+        // If we have a motion function, wrap it to forward the runtime params and pick `enter`.
+        // Otherwise, pass the `enter` motion object directly.
+        typeof value !== 'function' ? value.enter : (...args: Parameters<typeof value>) => value(...args).enter,
+      ),
+
+      // Wrap `exit` in its own motion component as a static method, e.g. <Fade.Out>
+      Out: createMotionComponent(
+        // If we have a motion function, wrap it to forward the runtime params and pick `exit`.
+        // Otherwise, pass the `exit` motion object directly.
+        typeof value !== 'function' ? value.exit : (...args: Parameters<typeof value>) => value(...args).exit,
+      ),
```

Comment:
nit let's invert the condition

```suggestion
      In: createMotionComponent(
        // If we have a motion function, wrap it to forward the runtime params and pick `enter`.
        // Otherwise, pass the `enter` motion object directly.
        typeof value === 'function' ? (...args: Parameters<typeof value>) => value(...args).enter : value.enter,
      ),

      // Wrap `exit` in its own motion component as a static method, e.g. <Fade.Out>
      Out: createMotionComponent(
        // If we have a motion function, wrap it to forward the runtime params and pick `exit`.
        // Otherwise, pass the `exit` motion object directly.
        typeof value === 'function' ? (...args: Parameters<typeof value>) => value(...args).exit : value.exit,
      ),
```